packages/editor/src/dataviews/store/private-actions.ts

```diff
@@ -113,6 +122,9 @@ export const registerPostTypeActions =
 				? duplicatePattern
 				: undefined,
 			postTypeConfig.supports?.title ? renamePost : undefined,
+			canManageOptions && postTypeConfig.slug === 'page'
```

Comment:
I see this logic as part of the editor package, so I presumed both the site editor & post editor would have the action. However, I've noticed the post editor **doesn't**:

| Site editor | Posts editor |
| --- | --- |
| <img width="277" alt="Screenshot 2024-11-14 at 10 06 28" src="https://github.com/user-attachments/assets/378b7470-065c-41cd-832c-35eeb9a8e9f3"> | <img width="277" alt="Screenshot 2024-11-14 at 10 06 03" src="https://github.com/user-attachments/assets/68329d6a-79be-4170-9694-3b5fc2623374"> |

It turns out to be that the `canManageOptions` check is undefined for the post editor. Would you know why is that?

Checked a few things and:

- the `canCreate` check works fine in the post editor
- the `canManageOptions` is actually resolved when using the post editor to edit a `post` (not a `page`)

🤔

Reply:
While [duplicating the action for the post editor](https://github.com/WordPress/gutenberg/pull/65426/commits/7a2960a713cb01258a26eba1a0620a6e2c14edf2), I've managed to get the new homepage action to show so hopefully that gets around the `canManageOptions` check being undefined, but `getEntityRecord( 'postType', 'page', [pageId] )` doesn't seem to work in this editor. I'm not sure why, it's like the editor only has access to the current page that's being edited.

Reply:
I think I've figured out a way to handle getting the current homepage in every context (page list, site editor and post editor). I'm not sure if it's the best way, but it seems to be working. I think the record was being checked too soon, so I've wrapped it in a `useEffect` in https://github.com/WordPress/gutenberg/pull/65426/commits/5e5cf0c5d7d792eb0de4fe4ffb440c0caf1a3536.